src/project/mod.rs

```diff
@@ -210,6 +224,24 @@ impl Project {
         Ok(project)
     }
 
+    /// Warns if Pixi is using a manifest from an environment variable rather than a discovered version
+    pub fn warn_on_discovered_from_env(manifest_path: Option<&Path>) {
+        if manifest_path.is_none() && std::env::var("PIXI_IN_SHELL").is_ok() {
+            let discover_path = find_project_manifest();
+            let env_path = std::env::var("PIXI_PROJECT_MANIFEST");
+
+            if let (Some(discover_path), Ok(env_path)) = (discover_path, env_path) {
+                if env_path.as_str() != discover_path.to_str().unwrap() {
+                    tracing::warn!(
+                        "Used manifest {} from `PIXI_PROJECT_MANIFEST` rather than local {}",
+                        env_path,
+                        discover_path.to_string_lossy()
+                    );
+                }
+            }
+        }
+    }
```

Comment:
I think this can just go right?

Reply:
I'm not sure exactly what you are asking, but there was a request on #1055 to have a warning after longer commands as well, so I have all of those calling this just before `OK(())`.

Reply:
I'm not a total fan of the warning that it gives but it's better then nothing. So lets leave it in and see if others have more feedback while using.